decentraland/screens/BuyLandScreen.js

```diff
@@ -53,16 +60,29 @@ export class BuyLandScreen extends React.Component {
 
     const onSuccess = () => {
       showInfo(`${typeDescription} bought successfully.`);
-      removeLandForSale(landForSale);
-      addToMyAssetsList(asset);
+    };
+
+    const onError = () => {
+      removeMyAssetFromList(asset);
+      addLandForSaleToList(landForSale);
```

Comment:
It's weird that the argument for these two functions isn't the exact same object or asset id. What's the difference between the contents of these two variables?

Reply:
The `landForSale` is the sell order and wraps the asset. For example:
```javascript
const assetForSale = {
  priceInWei,
  expiresAt,
  asset: {
    id, type, x, y
  }
}
```
The `MyAssetsScreen` has assets owned by the user instead of sell orders because it's no longer an open sell order.

Reply:
Ah ok right, that makes sense - it seems like there may be some clever way to DRY that up a bit, but I don't have a concrete suggestion so let's keep it as-is for now. Feel free to resolve this after reading.